tutorial/chapter/4/index.md

```diff
@@ -50,46 +42,27 @@ not 4
 
 ## Comparison operators
 
-The operators `<`, `<=`, `>`, and `>=` work like JavaScript, except they throw
-an error unless both values are strings or both values are numbers. It doesn't
-make much sense to ask if an array or an object is less than something else, so
-this helps avoid subtle bugs.
-
-The operator `==` is a deep equality operator. It works like JavaScript `===`
-for the purpose of comparising strings, numbers, `true`, `false`, `undefined`,
-`null`, and functions, but differs for arrays, and objects.
+The operators `<`, `<=`, `>`, and `>=` work like JavaScript, except they throw an error unless both values are strings or both values are numbers. It doesn't make much sense to ask if an array or an object is less than something else, so this helps avoid subtle bugs.
 
-For arrays, it checks to see if all elements of both arrays are equal to each
-other based on the Squiggle `==` operator. It is not aware of sparse arrays.
-
-For objects, it checks all keys returned by `Object.keys` on both objects for
-equality using the Squiggle `==` operator.
+The operator `==` is a strict equality operator. It works like JavaScript `===` for the purpose of comparing strings, numbers, `true`, `false`, `undefined`, and `null`, but it throws an exception if you try to compare objects, arrays, or functions. This is because those types do not have well-defined value comparison, so you should implement it yourself.
 
 As for `!=`, it's like `==` but returns the opposite value.
 
-```squiggle
-[] == []
-#=> true
-
-[2] == [1 + 1]
-#=> true
-
-["abc", "xyz"] == ["abc", "xy" ++ "z"]
-#=> true
+The operator `is` is equivalent to the ES6 function `Object.is`, [as documented
+on MDN][object_is]. Basically it's like JavaScript's `===` operator except that
+`0 is -0` is false, and `NaN is NaN` is true.
 
-{a: "b"} == {a: "b"}
-#=> true
+The operator `has` works like this is essentially like the following JavaScript function:
 
-{} == Object.create(null)
-#=> true
+```javascript
+function has(obj, key) {
+    if (obj === null || obj === undefined) throw new Error();
+    if (!validKey(key)) throw new Error();
+    return obj[key] !== undefined;
+}
 ```
 
-The operator `is` is equivalent to the ES6 function `Object.is`,
-[as documented on MDN][object_is].
-Basically it's like JavaScript's `===` operator except that
-`0 is -0` is false, and `NaN is NaN` is true.
-
-The operator `has` works like this:
+Valid keys are strings and whole numbers.
```

Comment:
i'm proper confused about [JavaScript Symbols](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol), mostly because of this:
```javascript
Symbol("foo") === Symbol("foo"); // false
```
which is entirely unlike ruby:
```ruby
:foo == :foo
# => true
```
clojure:
```clojure
(= 'foo 'foo)
;; => true                                                                                                                                    
(= :foo :foo)
;; => true
```